src/plugins/intel_gna/src/pre_post_process/preprocessing.hpp

```diff
@@ -6,37 +6,148 @@
 
 #include <cstdint>
 
+#include "avx2.hpp"
+#include "gna_data_types.hpp"
+
 namespace ov {
 namespace intel_gna {
 namespace pre_post_processing {
 
 void ConvertToInt16(int16_t* ptr_dst,
                     const float* ptr_src,
-                    const uint32_t num_rows,
-                    const uint32_t num_columns,
+                    const size_t num_rows,
+                    const size_t num_columns,
                     const float scale_factor);
 
 int16_t ConvertFloatToInt16(float src);
 int8_t ConvertFloatToInt8(float src);
 
-template <typename T1, typename T2>
-inline void UnscaleAndCast(T2* ptr_dst,
-                           T1* ptr_src,
-                           const uint32_t num_rows,
-                           const uint32_t num_columns,
-                           const float scale_factor) {
-    if (!ptr_dst || !ptr_src) {
+template <typename T, typename U>
+inline void unscale_transpose_and_cast(T* ptr_dst,
```

Comment:
I suggest follow the most used convention for method naming (PascalCase) even if SnakeCase seems to be more suitable for modern C++